sources/iTermBadgeLabel.m

```diff
@@ -99,9 +99,20 @@ - (NSImage *)freshlyComputedImage {
          _fillColor,
          NSStringFromSize(_viewSize),
          [NSThread callStackSymbols]);
+    float maxFont = [iTermAdvancedSettingsModel badgeMaxFontSize];
```

Comment:
Why not change the implementation of `-idealFontSize` since it already has a `min` and `max`? Then this becomes a much simpler change.